File: Other Algorithms/Kadane Algorithm/python/kadane.py
def kadane(a):
    max_current = max_global = a[0]
    for i in range(1, len(a)):
        max_current = max(a[i], max_current + a[i])
        if max_current > max_global:
            max_global = max_current
    return max_global

n = int(input())
a = [int(x) for x in input().split()]
print(kadane(a))

File: kadane.c
#include <stdio.h>
#define max(a, b) (a > b ? a : b)

int kadane(int *v, int n){
    int max_now = v[0], max_global = v[0];
    for(int i = 1; i < n; i++){
        max_now = max(max_now + v[i], v[i]);
        max_global = max(max_now, max_global);
    }
    return max_global;
}

int main(){
    int n;
    scanf("%d", &n);

    int v[n];

    for(int i = 0; i < n; i++){
        scanf("%d", &v[i]);
    }

    printf("%d", kadane(v, n));

}
